common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/responses/signin/SignInChallengeApiResponse.kt

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.providers.nativeauth.responses.signin
+
+import com.google.gson.annotations.Expose
+import com.google.gson.annotations.SerializedName
+import com.microsoft.identity.common.java.logging.LogSession
+import com.microsoft.identity.common.java.providers.nativeauth.IApiResponse
+import com.microsoft.identity.common.java.providers.nativeauth.interactors.InnerError
+import com.microsoft.identity.common.java.util.isOOB
+import com.microsoft.identity.common.java.util.isPassword
+import com.microsoft.identity.common.java.util.isRedirect
+import java.net.HttpURLConnection
+
+/**
+ * Represents the raw response from the /challenge endpoint.
+ * Can be converted to SignInChallengeApiResult using the provided toResult() method.
+ */
+data class SignInChallengeApiResponse(
+    @Expose override var statusCode: Int,
+    @Expose @SerializedName("credential_token") val credentialToken: String?,
+    @Expose @SerializedName("challenge_type") val challengeType: String?,
+    @Expose @SerializedName("binding_method") val bindingMethod: String?,
+    @Expose @SerializedName("challenge_target_label") val challengeTargetLabel: String?,
+    @Expose @SerializedName("challenge_channel") val challengeChannel: String?,
```

Comment:
Would challenge_target_label or challenge_channel correspond to a UPN? If so, I'm thinking they shouldn't be exposed?
Asking because exposed fields are logged, correct?

Reply:
Do we also need it removed for challenge channel? Seems like you only removed it for challenge target label

Reply:
challenge channel does not have any PII.